File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/tasklistview/ITaskListViewContract.java
package com.wiseassblog.samsaradayplanner.ui.tasklistview;

import com.wiseassblog.samsaradayplanner.domain.Tasks;

public interface ITaskListViewContract {
    interface View {
        void setTasks(Tasks tasks);
        void showMessage(String message);
        void navigateToDayView();
        void navigateToTaskView(int taskId);
    }

    interface ViewModel {
        void setTasks(Tasks tasks);
        Tasks getTasks();
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/tasklistview/ITaskListViewContract.kt
package com.wiseassblog.samsaradayplanner.ui.tasklistview

import com.wiseassblog.samsaradayplanner.common.Subscriber
import com.wiseassblog.samsaradayplanner.domain.Tasks

interface ITaskListViewContract {
    interface View {
        fun setTasks(tasks: Tasks)
        fun showMessage(message: String)
        fun navigateToDayView()
        fun navigateToTaskView(taskId: Int)
    }

    interface ViewModel {
        var tasks: Tasks
    }
}